### DISCUSSION TITLE
Access row data within formatter

### DISCUSSION THREAD

**Discussion by nmolinatcv** [grid-js/gridjs#925] [Q&A]
Is there a way to access the row data from within the formatter? I'm not interested in the row cells, especially since those are index based and any column changes would break those. Preferably, I'd like the formatter to have access to the "row" data found in "columns.data" function.

**Example**   
Let's assume I want a table that shows company name and score, but I want the name to be a link to the website and the score to have a data attribute with the company id. I'd like access to the row data object so I can easily reference the needed props by their name.
```js
// Imagine an array of companies with potentially 50 props per object
const companies = [
  { id: 1, name: 'Uber', website: 'uber.com', score: 5, ... },
  { id: 2, name: 'Twitter', website: 'twitter.com', score: 3, ... },
  ...
];

// I'd love a way to do the following using the formatter.
//   NOTE: The "_rowData" is made up, but it's what I am interested in.
//   Ideally I'd like it to be similar to the "row" object of the columns.data function.
new Grid({
  data: companies,
  columns: [
    {
      id: 'name',
      name: 'Company',
      formatter: (cell, row, col, _rowData) => {
        return html(`<a href="${_rowData.website}">${cell}</a>`);
      },
    },
    {
      id: 'score',
      name: 'Score',
      formatter: (cell, row, col, _rowData) => {
        return html(`<span data-id="${_rowData.id}">${cell}</span>`);
      },
    },
  ],
}).render(document.getElementById('company-scores-table');
```

**Comment by NazgulTuga** [comment]
Have you tried to add all the row data in the last position of the array?

`
const companies = [
  { id: 1, name: 'Uber', website: 'uber.com', score: 5, ..., {JSON_OBJECT} },
  { id: 2, name: 'Twitter', website: 'twitter.com', score: 3, ..., {JSON_OBJECT} },
  ...
];
`
See if the parameter "cell" in "formatter" has any data.
